src/prng.rs

```diff
@@ -7,51 +7,109 @@
 
 use super::field::{FieldElement, FieldError};
 use aes::{
-    cipher::{generic_array::GenericArray, FromBlockCipher, NewBlockCipher, StreamCipher},
+    cipher::{
+        generic_array::GenericArray, FromBlockCipher, NewBlockCipher,
+        StreamCipher as AesStreamCipher,
+    },
     Aes128, Aes128Ctr,
 };
 use getrandom::getrandom;
 
 use std::marker::PhantomData;
 
-const BLOCK_SIZE: usize = 16;
-const DEFAULT_BUFFER_SIZE_IN_ELEMENTS: usize = 128;
-const MAXIMUM_BUFFER_SIZE_IN_ELEMENTS: usize = 4096;
+const BUFFER_SIZE_IN_ELEMENTS: usize = 128;
 
-pub(crate) const SEED_LENGTH: usize = 2 * BLOCK_SIZE;
-
-pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, getrandom::Error> {
-    let mut seed = vec![0; SEED_LENGTH];
-    getrandom(seed.as_mut_slice())?;
+pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, PrngError> {
+    let mut seed = [0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
+    getrandom(&mut seed)?;
 
     // get prng array
-    let data: Vec<F> = Prng::new_with_seed_and_optional_length(&seed, Some(share1.len())).collect();
+    let data: Vec<F> = Prng::<_, Aes128Ctr>::new_with_seed(&seed)?
+        .take(share1.len())
+        .collect();
 
     // secret share
     for (s1, d) in share1.iter_mut().zip(data.iter()) {
         *s1 -= *d;
     }
 
-    Ok(seed)
+    Ok(seed.to_vec())
 }
 
 pub(crate) fn extract_share_from_seed<F: FieldElement>(
     length: usize,
     seed: &[u8],
 ) -> Result<Vec<F>, PrngError> {
-    if seed.len() != 2 * BLOCK_SIZE {
-        return Err(PrngError::SeedLen);
+    Ok(Prng::<_, Aes128Ctr>::new_with_seed(seed)?
+        .take(length)
+        .collect())
+}
+
+/// Generate a vector of uniform random field elements.
+pub fn random_vector<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
+    Ok(Prng::<_, Aes128Ctr>::new()?.take(len).collect())
+}
+
+/// A stream cipher maps a short key to a stream of pseudorandom bytes.
+pub(crate) trait StreamCipher: Sized {
+    const STREAM_CIPHER_SEED_SIZE: usize;
```

Comment:
`StreamCipher::STREAM_CIPHER_SEED_SIZE` repeats itself. I think `const SEED_SIZE: usize` suffices.

Reply:
`SEED_SIZE` is a bit ambiguous for `Aes128Ctr`, hence the prefix `STREAM_CIPHER`. Another way to disambiguate this is to define a StreamCipherAes128Ctr that implements StreamCipher, rather than implementing StreamCipher for Aes128Ctr directly.

Reply:
That's a good point, I hadn't thought about how it reads when accessed through an implementation of `StreamCipher`.